src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -518,6 +521,13 @@ private CreditCard buildCreditCard(final AdyenPaymentMethodsRecord paymentMethod
             creditCard.setCcSecCode(ccVerificationValue);
         }
 
+        final String ccUserAgent = PluginProperties.findPluginPropertyValue(PROPERTY_USER_AGENT, properties);
+        final String ccAcceptHeader = PluginProperties.findPluginPropertyValue(PROPERTY_ACCEPT_HEADER, properties);
+        if (ccUserAgent != null && ccAcceptHeader != null) {
```

Comment:
Yes, if one is missing it wouldn't trigger the 3DS process. And it is align with this: [CreditCardConverter.java#L111](https://github.com/killbill/killbill-adyen-plugin/blob/master/src/main/java/org/killbill/billing/plugin/adyen/client/payment/converter/impl/CreditCardConverter.java#L111).